driver-async/src/main/com/mongodb/async/client/MongoCollectionImpl.java

```diff
@@ -134,31 +139,31 @@ public ReadConcern getReadConcern() {
     @Override
     public <NewTDocument> MongoCollection<NewTDocument> withDocumentClass(final Class<NewTDocument> newDocumentClass) {
         return new MongoCollectionImpl<NewTDocument>(namespace, newDocumentClass, codecRegistry, readPreference, writeConcern, retryWrites,
-                retryReads, readConcern, executor);
+                retryReads, readConcern, uuidRepresentation, executor);
     }
 
     @Override
     public MongoCollection<TDocument> withCodecRegistry(final CodecRegistry codecRegistry) {
-        return new MongoCollectionImpl<TDocument>(namespace, documentClass, codecRegistry, readPreference, writeConcern, retryWrites,
-                retryReads, readConcern, executor);
+        return new MongoCollectionImpl<TDocument>(namespace, documentClass, createRegistry(codecRegistry, uuidRepresentation),
```

Comment:
We have to re-apply the UuidRepresentation at every level that a `CodecRegistry` can bet set.

Reply:
This is the only place I can foresee a possible issue. 

If I had a valid codecRegistry but now want to use and change the `TDocument` value and provide a new CodecRegistry users could hit an error.  The new TDocument type may not have a UUID in but it still could blow up.

I just wonder how Morphia and other wrapping libraries as well as users use these methods.

Reply:
I'm not clear on what the concern is.  Is it that someone would create their own implementation of `CodecRegistry` that doesn't use the `CodecRegistries` builder methods?  If so, I think that's not likely as it's rather complicated to build a proper implementation of that interface.  But it is a wart, and one we will fix in 4.0.

Reply:
I think my concern is probably unwarranted.